fix: migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API"

bdhirsh opened the pull request:
This is a re-land of https://github.com/pytorch/pytorch/pull/48097. The original regex that I added didn't correctly match namespaces that started with an underscore (e.g. `_test`), which caused a master-only test to fail.

The only change from the previous commit is that I updated the regex like so:

before: `^.*TORCH_LIBRARY_IMPL_init_([^_]+)_([^_]+)_[0-9]+(\(.*)?$`
after: `^.*TORCH_LIBRARY_IMPL_init_([_]*[^_]+)_([^_]+)_[0-9]+(\(.*)?$`

I added in a `[_]*` to the beginning of the namespace capture. I did the same for the `_FRAGMENT` regex.

Verified that running `ANALYZE_TEST=1 tools/code_analyzer/build.sh` (as the master-only test does) produces no diff in the output.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48098 make duplicate def() calls an error in the dispatcher
* #48178 Updating all call-sites of the legacy dispatcher registration API in fbcode to the new API.
* **#48308 fix: migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API"**

Fixing regex pattern to allow for underscores at the beginning of the
namespace

This reverts commit 3c936ecd3c68f395dad01f42935f20ed8068da02.

Differential Revision: [D25123295](https://our.internmc.facebook.com/intern/diff/D25123295)

bdhirsh committed:
Revert "Revert D25056091: migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API"

Fixing regex pattern to allow for underscores at the beginning of the
namespace

This reverts commit 3c936ecd3c68f395dad01f42935f20ed8068da02.

[ghstack-poisoned]

bdhirsh committed:
Update on "Revert "Revert D25056091: migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API""

The original regex that I added didn't correctly match namespaces that started with an underscore (e.g. `_test`), which caused a master-only test to fail.

The only change from the previous commit is that I updated the regex like so:

before: `^.*TORCH_LIBRARY_IMPL_init_([^_]+)_([^_]+)_[0-9]+(\(.*)?$`
after: `^.*TORCH_LIBRARY_IMPL_init_([_]*[^_]+)_([^_]+)_[0-9]+(\(.*)?$`

I added in a `[_]*` to the beginning of the namespace capture. I did the same for the `_FRAGMENT` regex.

Verified that running `ANALYZE_TEST=1 tools/code_analyzer/build.sh` (as the master-only test does) produces no diff in the output.


Fixing regex pattern to allow for underscores at the beginning of the
namespace

This reverts commit 3c936ecd3c68f395dad01f42935f20ed8068da02.

Differential Revision: [D25123295](https://our.internmc.facebook.com/intern/diff/D25123295)

[ghstack-poisoned]

ezyang reviewed (approved):
You can run master only tests by pushing to a branch named `ci-all/something`

bdhirsh committed:
Update on "fix: migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API""

This is a re-land of https://github.com/pytorch/pytorch/pull/48097. The original regex that I added didn't correctly match namespaces that started with an underscore (e.g. `_test`), which caused a master-only test to fail.

The only change from the previous commit is that I updated the regex like so:

before: `^.*TORCH_LIBRARY_IMPL_init_([^_]+)_([^_]+)_[0-9]+(\(.*)?$`
after: `^.*TORCH_LIBRARY_IMPL_init_([_]*[^_]+)_([^_]+)_[0-9]+(\(.*)?$`

I added in a `[_]*` to the beginning of the namespace capture. I did the same for the `_FRAGMENT` regex.

Verified that running `ANALYZE_TEST=1 tools/code_analyzer/build.sh` (as the master-only test does) produces no diff in the output.


Fixing regex pattern to allow for underscores at the beginning of the
namespace

This reverts commit 3c936ecd3c68f395dad01f42935f20ed8068da02.

Differential Revision: [D25123295](https://our.internmc.facebook.com/intern/diff/D25123295)

[ghstack-poisoned]

bdhirsh committed:
Update on "fix: migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API""

This is a re-land of https://github.com/pytorch/pytorch/pull/48097. The original regex that I added didn't correctly match namespaces that started with an underscore (e.g. `_test`), which caused a master-only test to fail.

The only change from the previous commit is that I updated the regex like so:

before: `^.*TORCH_LIBRARY_IMPL_init_([^_]+)_([^_]+)_[0-9]+(\(.*)?$`
after: `^.*TORCH_LIBRARY_IMPL_init_([_]*[^_]+)_([^_]+)_[0-9]+(\(.*)?$`

I added in a `[_]*` to the beginning of the namespace capture. I did the same for the `_FRAGMENT` regex.

Verified that running `ANALYZE_TEST=1 tools/code_analyzer/build.sh` (as the master-only test does) produces no diff in the output.


Fixing regex pattern to allow for underscores at the beginning of the
namespace

This reverts commit 3c936ecd3c68f395dad01f42935f20ed8068da02.

Differential Revision: [D25123295](https://our.internmc.facebook.com/intern/diff/D25123295)

[ghstack-poisoned]
